compiler/qsc/src/interpret.rs

```diff
@@ -468,6 +466,38 @@ impl Interpreter {
         ))
     }
 
+    fn run_with_sim_no_output(
+        &mut self,
+        entry_expr: Option<String>,
+        sim: &mut impl Backend<ResultType = impl Into<val::Result>>,
+    ) -> InterpretResult {
+        let mut sink = std::io::sink();
+        let mut out = GenericReceiver::new(&mut sink);
+
+        let (package_id, graph) = if let Some(entry_expr) = entry_expr {
+            // entry expression is provided
+            (self.package, self.compile_entry_expr(&entry_expr)?.0.into())
+        } else {
+            // no entry expression, use the entrypoint in the package
+            (self.source_package, self.get_entry_exec_graph()?)
+        };
+
+        if self.quantum_seed.is_some() {
+            sim.set_seed(self.quantum_seed);
+        }
+
+        eval(
+            package_id,
+            self.classical_seed,
```

Comment:
So this doesn't override the seed that might have been set by line 486 above?

Reply:
(I guess this is specifically the classical seed, so no)

Reply:
Correct. A bit clunky, but I'm copying the pattern we've already established in 2 other places in this file we call `set_seed`. This is getting to the point where it needs a refactor again...